### PR TITLE
Fix for RavenDB-13499

### PR DESCRIPTION
**Pull request by myarichuk**
* Support for alias in edge SELECT clauses so the following queries would work:
match (Person as f)-[Relationships as r select __*r.TargetId*__]->(Person as t)
* Improve unit test asserts for RavenDB-11139

### CODE REVIEW

File: src/Raven.Client/Documents/Operations/CompareExchange/CompareExchangeValueResultParser.cs [hunk 1]
```diff
@@ -59,8 +61,10 @@ public static CompareExchangeValue<T> GetValue(BlittableJsonReaderObject respons
             if (response == null)
                 return null;
 
-            var value = GetValues(response, conventions).FirstOrDefault();
+            var compareExchangeValues = GetValues(response, conventions);
+            var value = compareExchangeValues.FirstOrDefault();
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/CompareExchange/CompareExchangeValueResultParser.cs]: Looks like debugging code and unrelated to the issue?

File: test/SlowTests/Server/Documents/PeriodicBackup/RavenDB-11139.cs [hunk 2]
```diff
@@ -31,7 +32,12 @@ public async Task CreateFullAndIncrementalBackupWithCompareExchange()
                 {
                     Name = "💩"
                 };
-                await store.Operations.SendAsync(new PutCompareExchangeValueOperation<User>("emojis/poo", user, 0));
+
+                var operationResult = await store.Operations.SendAsync(new PutCompareExchangeValueOperation<User>("emojis/poo", user, 0));
```
Comment (ppekrol) [inline test/SlowTests/Server/Documents/PeriodicBackup/RavenDB-11139.cs]: Unrelated to the issue?

File: src/Raven.Client/Documents/Operations/CompareExchange/CompareExchangeValueResultParser.cs (hunk 1 above)
Comment (myarichuk) [reply src/Raven.Client/Documents/Operations/CompareExchange/CompareExchangeValueResultParser.cs]: yes, it is unrelated indeed.

File: test/SlowTests/Server/Documents/PeriodicBackup/RavenDB-11139.cs (hunk 2 above)
Comment (myarichuk) [reply test/SlowTests/Server/Documents/PeriodicBackup/RavenDB-11139.cs]: yes, it is unrelated indeed.

**Comment by myarichuk** [comment]
note: removed code unrelated to the commit

**Pull request merged**